### PR TITLE
Feature/get friends and all users displaying

### PR DESCRIPTION
**Pull request by nataliemcintyre2021**
# Thank you for pushing your code!
## What (if any) features are you implementing?
- Query set up for getAllUsers
- Functionality working for UserDisplay component (which renders, Search, DisplayList, Cards)
- When user visits UserDisplay page, search bar will appear along with how many friends they have with their friends listed below. Upon change in search bar, all users will be filtered to allow users to search for new friends

## What (if anything) did you refactor?
- UserDisplay, Search, DisplayList, Cards and minimal styling for each

## Were there any issues that arose?
- Accounting for users/friends who have null types (added logic in Cards to account for this)
- **The allUsers seems to include the logged - in user (?) so they seem to be appearing in searches**
- **Searching searches by if any username includes a character types - may want to rethink if all of those are displaying (?)**

## Is there anything that you need from your teammates?
- 

## Any other comments, questions, or concerns?
- Needs styling, testing, router

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor

## How Has This Been Tested?
- [ ] open index.html
- [X] console.log()
- [X] dev tools
- [ ] Cypress

## Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/DisplayList/DisplayList.js [hunk 1]
```diff
@@ -1,42 +1,44 @@
...
+import './DisplayList.css';
 
-const DisplayList = ({ friends, allUsers, searchActivated }) => {
+const DisplayList = ({ friends, filteredUsers}) => {
 
   const friendsCards = friends.map(friend => {
     return (
       <Cards
-        icon={friend.icon}
+        key={friend.id}
```
Comment (MTCawthray) [inline src/components/DisplayList/DisplayList.js:10]: good call on making the ID the key.

File: src/components/DisplayList/DisplayList.js [hunk 2]
```diff
@@ -1,42 +1,44 @@
...
-    )
-  })
-  const allUsersCards = allUsers.map(user => {
-    return (
-      <Cards
-        icon={user.icon}
-        username={user.username}
-        myersBriggs={user.myersBriggs}
-        enneagram={user.enneagram}
+        myersBriggs={friend.myersBrigg ? friend.myersBrigg.typeOf : null}
```
Comment (MTCawthray) [inline src/components/DisplayList/DisplayList.js:12]: Thank you for adding this error handling! We can come up with some sort of icon or something to display when null is the type!

File: src/components/DisplayList/DisplayList.js [hunk 3]
```diff
@@ -1,42 +1,44 @@
...
+        enneagram={friend.enneagram ? friend.enneagram.number : null}
       />
     )
   })
 
+    const allUsersCards = (filteredUsers && filteredUsers.map(user => {
```
Comment (MTCawthray) [inline src/components/DisplayList/DisplayList.js:18]: I'm loving the use of the conditional rendering for the filtered users.

File: src/components/DisplayList/DisplayList.js [hunk 4]
```diff
@@ -1,42 +1,44 @@
...
-        {allUsersCards}
-      </section> :
-      <section className="show-friends">
-        <h3>{friends.length} friends</h3>
-        <section className="friendsCards">
-          {friendsCards}
-        </section>
-      </section>
-    }
+    { filteredUsers ?
```
Comment (MTCawthray) [inline src/components/DisplayList/DisplayList.js:31]: Great ternary use case!

File: src/components/UserDisplay/UserDisplay.js [hunk 5]
```diff
@@ -1,39 +1,45 @@
...
   const [allUsers, setAllUsers] = useState([]);
-  const [searchActivated, setSearchActivated] = useState(false)
-  const [filteredUsers, setFilteredUsers] = useState()
+  const [filteredUsers, setFilteredUsers] = useState(null)
+  const [searchInput, setSearchInput] = useState('')
+  const {error, loading, data} = useQuery(GET_ALL_USERS)
+
+
+  useEffect(() => {
+    (!loading && setAllUsers(data.getAllUsers));
```
Comment (MTCawthray) [inline src/components/UserDisplay/UserDisplay.js:17]: The !loading feels like a nice async work around. It's helpful that we are using it all over our repo so when someone sees it once, they'll know whats happening elsewhere!

File: src/graphQL/queries.js [hunk 6]
```diff
@@ -39,3 +39,25 @@ export const GET_USER = gql `
       }
     }
   }`
+
+  export const GET_ALL_USERS = gql `
```
Comment (MTCawthray) [inline src/graphQL/queries.js:43]: Thanks to you and Gunnar for getting this query set up!

**Review by MTCawthray — APPROVED** [review]
This looks great. It's a huge step towards our MVP, so thank you for taking this on!

File: src/components/App/App.js [hunk 7]
```diff
@@ -31,9 +32,10 @@ function App() {
 
   return (
     <div className="App">
-      
-      <Login />
-  
+      <Header />
+
+      {!loading && <UserDisplay friends={data.getUserStats.friends} />}
```
Comment (hayleyw7) [inline src/components/App/App.js:37]: The Router PR also adds a line like this, so we'll have to see how they all merge together to get it all seamless.

File: src/components/DisplayList/DisplayList.js (hunk 1 above)
Comment (hayleyw7) [reply src/components/DisplayList/DisplayList.js:10]: ^^ agree with Mark

File: src/components/DisplayList/DisplayList.js (hunk 2 above)
Comment (hayleyw7) [reply src/components/DisplayList/DisplayList.js:12]: ^ Natalie, I just asked Mark about this, and he brought up displaying `...`, `?`, or something else to show they haven't yet set up their profile.

Another idea would be not letting a user exist to others at all until they save their profile, with all fields being mandatory (username, enneagramType, mbtiType). I think I prefer this idea myself, but I value everyone's thoughts on this, and either would work.

**Review by hayleyw7 — APPROVED** [review]

**Pull request merged**